src/main/scripts/createAppGatewayIngress.sh

```diff
@@ -52,31 +52,52 @@ function network_peers_aks_appgw() {
 
     aksNetWorkId=$(az resource list -g ${aksMCRGName} --resource-type Microsoft.Network/virtualNetworks -o tsv --query '[*].id')
     aksNetworkName=$(az resource list -g ${aksMCRGName} --resource-type Microsoft.Network/virtualNetworks -o tsv --query '[*].name')
-    az network vnet peering create \
-        --name aks-appgw-peer \
-        --remote-vnet ${aksNetWorkId} \
-        --resource-group ${curRGName} \
-        --vnet-name ${appgwVNetName} \
-        --allow-vnet-access
-    utility_validate_status "Create network peers for $aksNetWorkId and ${appgwVNetName}."
-
-    appgwNetworkId=$(az resource list -g ${curRGName} --name ${appgwVNetName} -o tsv --query '[*].id')
-    az network vnet peering create \
-        --name aks-appgw-peer \
-        --remote-vnet ${appgwNetworkId} \
-        --resource-group ${aksMCRGName} \
-        --vnet-name ${aksNetworkName} \
-        --allow-vnet-access
-
-    utility_validate_status "Create network peers for $aksNetWorkId and ${appgwVNetName}."
+    appGatewaySubnetId=$(az network application-gateway show -g ${curRGName} --name ${appgwName} -o tsv --query "gatewayIpConfigurations[0].subnet.id")
+    appGatewayVnetResourceGroup=$(az network application-gateway show -g ${curRGName} --name ${appgwName} -o tsv --query "gatewayIpConfigurations[0].subnet.resourceGroup")
+    appGatewaySubnetName=$(az resource show --ids ${appGatewaySubnetId} --query "name" -o tsv)
+    appgwNetworkId=$(echo $appGatewaySubnetId | sed s/"\/subnets\/${appGatewaySubnetName}"//)
+    appgwVnetName=$(az resource show --ids ${appgwNetworkId} --query "name" -o tsv)
+
+    local toPeer=true
+    # if the AKS and App Gateway have the same VNET, need not peer.
+    if [ "${aksNetWorkId}" == "${appgwNetworkId}" ]; then
```

Comment:
I have asked this general question in several code reviews: why does this script not use the important best practices for shell scripting, such as those in [https://transang.me/best-practice-to-make-a-shell-script/](https://transang.me/best-practice-to-make-a-shell-script/)? 

In particular 

```
#!/usr/bin/env bash
set -Eeuo pipefail
```

If we had that, I would worry less about the many invocations of sub commands such as the `az network vnet` one below and elsewhere.

Reply:
I will accept this now, but I would like to understand why we don't need the best practices here.

Reply:
Good point. Here is some of background why some of these flags are not applied. For example, we want to output explicit and meaningful error message if script failed, which is helpful for user to trouble shoot the issue. Another example is that some of command invoked in the script may return non-zero value, but it doesn't mean it's a fatal error that the execution should exit. 

I do agree applying best practice of shell programming is meaningful, so https://github.com/WASdev/azure.liberty.aks/issues/50 is created for tracking the requirement of enhancement and let's move forward as for now.